docker/Makefile

```diff
@@ -127,11 +128,24 @@ TEST_FILES += -f services_base.yaml
 TEST_FILES += -f ccf_base.yaml
 TEST_FILES += -f test.yaml
 
+DOCKER_COMPOSE_COMMAND=docker-compose
+
+ifeq ($(SGX_MODE),HW)
+	TEST_FILES += -f test-sgx.yaml
+	SGX_DEVICE_PATH=$(shell if [ -e "/dev/isgx" ]; \
+					then echo "/dev/isgx"; \
+					elif [ -e "/dev/sgx/enclave" ]; \
+						then echo "/dev/sgx/enclave"; \
+					else echo "ERROR: NO SGX DEVICE FOUND"; \
+					fi)
+	DOCKER_COMPOSE_COMMAND := env SGX_MODE=$(SGX_MODE) SGX_DEVICE_PATH=${SGX_DEVICE_PATH} ${DOCKER_COMPOSE_COMMAND}
+endif
+
 build_test : repository build_services build_ccf build_client
```

Comment:
is there any reason why we don't just have a separate target for building with sgx? then set the hw mode explicitly. you're assumption that it is set outside the makefile (but the devices aren't) seems a bit complicated.

Reply:
hmm, but isn't that the same we do with ledger and interpreter type where we also assume they are externally defined?  Seems to me easier to get it everything consistent with single parameterized targets without having to define separate targets for all permutations?

Reply:
we NEVER change those. the build args always default. (and so, if you want to change those, i'm ok with a little more complexity). the HW mode is a lot more interesting.

Reply:
i don't really see what the complexity is with the current approach: duplicating multiple targets (at least build and test variants and potentially also some of the run/stop targets)  is not really easier and imho harder to maintain?  But i definitely second the proposal to "type" the names of the images as otherwise there is a decent risk that get into some mix'n'(mis)match situation which could lead to unpredictable (and maybe unintuitive) errors?

Reply:
We have always selected our build and tests in hw/sim mode based on the `SGX_MODE` environment variable. In fact, the variable is also documented in `docs/install.md` since it's user facing.

Now, from today's discussion, the idea is to have separate targets for build/test in hw/sim mode. To make things clean, this means that `SGX_MODE` should really become an internal variable, not set externally by default (or, in any case, it should be always overridden), and removed from the docs.

Reply:
I have started doing some work to have different targets and typed images here https://github.com/bvavala/private-data-objects/tree/bruno.240315.specialize-images .
The branch still does not work and it's already too complicated. We have multiple interdependent images. We have to change: makefile, yaml and docker files.

I think the PR as it is already fixes the mix'n'(mis)match situations since SGX_MODE is built into the images -- you can simply check that.

Reply:
Would you please take a look at the second commit in this PR?

It adds the separate targets. It does _not_ type the images (except setting the SGX_MODE var inside). And it removes the SGX_MODE var as a user/dev-facing variable -- so sim/hw-mode selection is all done explicitly through the makefile.